samples/net/sockets/echo_async_select/src/socket_echo_select.c

```diff
@@ -124,7 +124,9 @@ int main(void)
 	res = setsockopt(serv6, IPPROTO_IPV6, IPV6_V6ONLY, &TRUE, sizeof(TRUE));
 	if (res < 0) {
 		printf("error: setsockopt: %d\n", errno);
+#ifndef __ZEPHYR__
```

Comment:
So, how it worked previously is that there's `#ifdef IPV6_V6ONLY` at line 119, and that was not defined for Zephyr, so the whole code block wasn't compiled for Zephyr. Indeed, I added definition of IPV6_V6ONLY recently, and there was coverity error check fix recently, which looks like effectively broke this sample.

But I'd suggest to fix it in a way to not sprinkle more ifdef's around. Either:

1. At line 119, replace `#ifdef IPV6_V6ONLY` with `#ifndef __ZEPHYR__`.
2. (Better) to let setsockopt to return successful code for IPV6_V6ONLY.

Reply:
@pfalcon 

> (Better) to let setsockopt to return successful code for IPV6_V6ONLY

I think, this would create an obscure puzzle for somebody later.

Anyway, tweaked it slightly, please take a look.

Reply:
> I think, this would create an obscure puzzle for somebody later.

It shouldn't be much of such, because IPV6_V6ONLY is the default Zephyr behavior. And we have such cases already, and going to have more: https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/net/lib/sockets/sockets.c#L1117 . There're just too many adhoc and obscure socket options exist, and we either allow to run existing (3rd-party) apps with Zephyr, or disallow that, allowing apparently makes more sense.